### DISCUSSION TITLE
How to convert between different formats

### DISCUSSION THREAD

**Discussion by egemengol** [Serial-ATA/lofty-rs#223] [Q&A] [answered]
Hi, I'm an intermediate Rust developer who knows nothing about audio.

I'm writing a tool which talks to multiple APIs:
- Telegram API recognises MP3 and M4A
- Google Speech2Text recognises WAV

I want to convert back and forth, can I utilize lofty for this purpose? Can anyone point me to the right direction?

Thanks in advance :)

**Comment by Serial-ATA** [comment, marked as answer]
Hello!

Lofty is only able to convert metadata between formats. To convert the actual audio you'll need another crate. You'd probably be best off using FFmpeg, either as a command or through the [ffmpeg-next crate](https://crates.io/crates/ffmpeg-next). There currently aren't any great pure Rust solutions for this.

**Comment by egemengol** [comment]
Thanks :)
